pennylane/capture/capture_qnode.py

```diff
@@ -53,34 +93,102 @@ def _get_shapes_for(*measurements, shots=None, num_device_wires=0):
     for s in shots:
         for m in measurements:
             shape, dtype = m.aval.abstract_eval(shots=s, num_device_wires=num_device_wires)
-            shapes.append(jax.core.ShapedArray(shape, dtype_map.get(dtype, dtype)))
+            shapes.append(jax.core.ShapedArray(batch_shape + shape, dtype_map.get(dtype, dtype)))
+
     return shapes
 
 
 @lru_cache()
 def _get_qnode_prim():
+
     if not has_jax:
         return None
+
     qnode_prim = jax.core.Primitive("qnode")
     qnode_prim.multiple_results = True
 
     # pylint: disable=too-many-arguments
     @qnode_prim.def_impl
-    def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts):
+    def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts, batch_dims=None):
         consts = args[:n_consts]
-        args = args[n_consts:]
+        non_const_args = args[n_consts:]
 
         def qfunc(*inner_args):
             return jax.core.eval_jaxpr(qfunc_jaxpr, consts, *inner_args)
 
         qnode = qml.QNode(qfunc, device, **qnode_kwargs)
-        return qnode._impl_call(*args, shots=shots)  # pylint: disable=protected-access
+
+        if batch_dims is not None:
+            # pylint: disable=protected-access
+            return jax.vmap(partial(qnode._impl_call, shots=shots), batch_dims)(*non_const_args)
+
+        # pylint: disable=protected-access
+        return qnode._impl_call(*non_const_args, shots=shots)
 
     # pylint: disable=unused-argument
     @qnode_prim.def_abstract_eval
-    def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts):
+    def _(*args, qnode, shots, device, qnode_kwargs, qfunc_jaxpr, n_consts, batch_dims=None):
+
         mps = qfunc_jaxpr.outvars
-        return _get_shapes_for(*mps, shots=shots, num_device_wires=len(device.wires))
+
+        return _get_shapes_for(
+            *mps,
+            shots=shots,
+            num_device_wires=len(device.wires),
+            batch_shape=_get_batch_shape(args[n_consts:], batch_dims),
+        )
+
+    # pylint: disable=too-many-arguments, too-many-positional-arguments
+    def _qnode_batching_rule(
+        batched_args,
+        batch_dims,
+        qnode,
+        shots,
+        device,
+        qnode_kwargs,
+        qfunc_jaxpr,
+        n_consts,
+    ):
+        """
+        Batching rule for the ``qnode`` primitive.
+
+        This rule exploits the parameter broadcasting feature of the QNode to vectorize the circuit execution.
```

Comment:
This docstring must be updated after the last few commits (now we are calling `jax.vmap` in the definition implementation). I'll do that in the next commit